src/writer.rs

```diff
@@ -118,19 +113,17 @@ impl Writer {
     pub fn write_from_iter<I>(&mut self, streamline: I, len: usize)
         where I: IntoIterator<Item = Point>
     {
-        self.writer.write_i32::<LittleEndian>(len as i32).unwrap();
-        for p in streamline {
-            self.write_point_and_scalars(&p, &[]);
-        }
-        self.real_n_count += 1;
+        write_streamline!(self, streamline, len);
     }
 
-    fn write_point_and_scalars(&mut self, p: &Point, scalars: &[f32]) {
+    fn write_point(&mut self, p: &Point) {
         let p = self.affine * p + self.translation;
         self.writer.write_f32::<LittleEndian>(p.x).unwrap();
         self.writer.write_f32::<LittleEndian>(p.y).unwrap();
         self.writer.write_f32::<LittleEndian>(p.z).unwrap();
+    }
 
+    fn write_sp(&mut self, scalars: &[f32]) {
```

Comment:
You're only writing the `s` in `sp` here. Rename?

Reply:
My bad. `write_sp` is used to write 's' then 'p', the signature should be 

    fn write_sp(&mut self, data: &[f32]) {

Or `write_f32s` or something. Do you have a suggestion?

Reply:
`write_f32s` is fine